workitems/tokenlist/l2-token-list-v1.0-psd01.md

```diff
@@ -0,0 +1,685 @@
+**[O]** - The requirement level for requirements which IDs start with the letter _O_ is to be interpreted as **MAY** as described in RFC2119. 
+
+Note that requirements are uniquely numbered in ascending order within each requirement level.
+
+Example : It should be read that [R1] is an absolute requirement of the specification whereas [D1] is a recommendation and [O1] is truly optional.
+
+
+-------
+
+# 2 Concept and Design
+
+For lists of canonical tokens, L2s currently maintain their own customized versions of the Uniswap token list. For example, Arbitrum maintains a [token list](https://bridge.arbitrum.io/token-list-42161.json) with various custom extensions. Optimism also maintains a [custom token list](https://raw.githubusercontent.com/ethereum-optimism/ethereum-optimism.github.io/master/optimism.tokenlist.json), but with different extensions. It should be noted that both of these custom extensions refer to the bridge that these tokens can be carried through. However, these are not the only bridges that the tokens can be carried through, which means that bridges and token lists should be separated. Also note that currently, both Optimism and Arbitrum base "canonicity" on the token name + symbol pair.
+
+An example of an Arbitrum token entry is given below:
+```
+{
+logoURI: "https://assets.coingecko.com/coins/images/13469/thumb/1inch-token.png?1608803028",
+chainId: 42161,
+address: "0x6314C31A7a1652cE482cffe247E9CB7c3f4BB9aF",
+name: "1INCH Token",
+symbol: "1INCH",
+decimals: 18,
+extensions: {
+  bridgeInfo: {
+    1: {
+    tokenAddress: "0x111111111117dc0aa78b770fa6a738034120c302",
+    originBridgeAddress: "0x09e9222e96e7b4ae2a407b98d48e330053351eee",
+    destBridgeAddress: "0xa3A7B6F88361F48403514059F1F16C8E78d60EeC"
+     }
+   }
+  }
+}
+```
+This standard will build upon the current framework and augment it with concepts from the W3C DID Specification [[1]](#w3c-did) based on the JSON linked data model [[2]](#jsonld) such as resolvable unique resource identifiers (URIs) and JSON-LD schemas which enable easier schema verification using existing tools.
+
+
+-------
+
+# 3 Token List Specification
+
+The schema for a canonical token list utilizes draft version 7 of https://json-schema.org for consistency purposes with the [W3C CCG](https://w3c-ccg.github.io/) effort. 
+
+#### **[R1]**
+The following data elements MUST be present in a canonical token list:
+- type
+- tokenListId
+- name
+- createdAt
+- updatedAt
+- versions
+- tokens
+
+Note, that the detailed definition of the data elements in [[R1]](#r1) along with descriptions and examples are given in the schema itself below.
+
+#### **[R2]**
+The tokens data element is a composite which MUST minimally contain the following data elements:
+- chainId
+- chainURI
+- tokenId
+- tokenType
+- address
+- tokenName
+- tokenSymbol
+- decimals
```

Comment:
erc 777 refers to decimals as 'granularity' this document should state somewhere the equivalency

Reply:
@realcodywburns will include in the description of the `decimals` property in the JSON schema.